sentry/src/test/java/io/sentry/config/FilesystemPropertiesLoaderTest.kt

```diff
@@ -0,0 +1,51 @@
+package io.sentry.config
+
+import io.sentry.NoOpLogger
+import java.nio.charset.Charset
+import kotlin.test.AfterTest
+import kotlin.test.BeforeTest
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertNotNull
+import kotlin.test.assertNull
+import org.junit.rules.TemporaryFolder
+
+class FilesystemPropertiesLoaderTest {
+
+    var folder = TemporaryFolder()
+
+    @BeforeTest
+    fun `create temporary folder`() {
+        folder.create()
+    }
+
+    @AfterTest
+    fun `delete temporary folder`() {
+        folder.delete()
```

Comment:
this will fail if the folder is not empty, kotlin has an extension to delete a folder recursively.
`file. deleteRecursively()`

Reply:
It would work in this case as `delete()` was called on the class that was meant to be used as JUnit 4 rule. Now since you've shown me how to use rules in Kotlin it's not an issue anymore.